
**Diplomatura en Ciencia de Datos, Aprendizaje Automático y sus Aplicaciones**

**Edición 2022**

---

# Trabajo práctico entregable - Parte 2

In [84]:
import io
import matplotlib
import matplotlib.pyplot as plt
import numpy
import pandas as pd
import seaborn
import math
import scipy as sp
from scipy.stats import t
import scipy.stats 
import numpy as np
seaborn.set_context('talk')

## Lectura del dataset

En la notebook 00 se explican los detalles de la siguiente sección.

In [85]:
url = 'https://cs.famaf.unc.edu.ar/~mteruel/datasets/diplodatos/sysarmy_survey_2020_processed.csv'
df = pd.read_csv(url)

In [86]:
df[:3]

profile_gender  profile_age work_country                    work_province  \
0          Mujer           26    Argentina  Ciudad Autónoma de Buenos Aires   
1         Hombre           29    Argentina                       Corrientes   
2          Mujer           22    Argentina  Ciudad Autónoma de Buenos Aires   

   profile_years_experience  work_years_in_company  \
0                       3.0                    3.0   
1                       5.0                    2.0   
2                       2.0                    0.0   

   work_years_in_current_position  work_people_in_charge_of  \
0                             3.0                         0   
1                             2.0                         4   
2                             0.0                         0   

  profile_studies_level profile_studies_level_state  ...  \
0         Universitario                    En curso  ...   
1         Universitario                    En curso  ...   
2            Secundario                  Completado  ...   

  work_has_violence_situations profile_sexual_orientation  \
0         En mi trabajo actual                 Homosexual   
1                        Jamás               Heterosexual   
2       En un trabajo anterior           Bisexual o queer   

  profile_has_disabilities profile_has_disabilities_hiring_difficulties  \
0                      NaN                                          NaN   
1                   Visual                                           No   
2                      NaN                                           No   

  company_employee_number                          company_main_activity  \
0                501-1000  Servicios / Consultoría de Software / Digital   
1                 201-500                               Otras industrias   
2               2001-5000                               Otras industrias   

  company_recommended company_diversity_policies  \
0                   7                          2   
1                   8                          9   
2                   6                          9   

                              company_extra_benefits  \
0  Capacitaciones y/o cursos, Comidas pagas / sub...   
1  Horarios flexibles, Stock options / RSUs, Viát...   
2  Clases de gimnasia online, Comidas pagas / sub...   

  company_best_companies_city  
0                         NaN  
1                         NaN  
2                         NaN  

[3 rows x 48 columns]

In [87]:
df[['profile_gender', 'salary_monthly_NETO']].groupby('profile_gender').describe()

salary_monthly_NETO                                     \
                             count          mean            std   min   
profile_gender                                                          
Hombre                      4944.0  98836.063558  102159.664237   1.0   
Mujer                        908.0  73501.414465   56256.880181   2.0   
Otros                         31.0  95441.354839   88635.674358  40.0   

                                                       
                    25%      50%       75%        max  
profile_gender                                         
Hombre          53285.0  80000.0  110000.0  2080000.0  
Mujer           43818.0  65000.0   92000.0   800000.0  
Otros           39500.0  74000.0  117500.0   380000.0

In [88]:
df[df.salary_monthly_NETO > 1000]\
  [['profile_gender', 'salary_monthly_NETO']].groupby('profile_gender').describe()

salary_monthly_NETO                                         \
                             count           mean            std      min   
profile_gender                                                              
Hombre                      4815.0  101481.624343  102215.304684   1100.0   
Mujer                        863.0   77328.913778   55082.760570   2000.0   
Otros                         28.0  105660.714286   87235.710235  34000.0   

                                                       
                    25%      50%       75%        max  
profile_gender                                         
Hombre          55000.0  80000.0  110000.0  2080000.0  
Mujer           47850.0  67200.0   93000.0   800000.0  
Otros           43750.0  78900.0  122750.0   380000.0

In [89]:
alpha = 0.05

In [90]:
is_man = df.profile_gender == 'Hombre'

groupA = df[(df.salary_monthly_NETO > 1000) & is_man].salary_monthly_NETO
groupB = df[(df.salary_monthly_NETO > 1000) & ~is_man].salary_monthly_NETO

## Ejercicio 1: Estimación

**Consigna:**  Calcular una estimación puntual y un intervalo de confianza de nivel (1-alpha) para la resta entre la media del salario Neto para Hombres menos la media del salario Neto para otros géneros(diferencia de las medias entre el grupoA y grupoB). 
¿Cómo se relaciona este intervalo de confianza con el test de hipótesis?

### Solución

El parámetro a estimar es la diferencia de medias. Se calculan las medias de los dos grupos de variables y
luego se efectua la resta

Suponiendo que las muestras empleados en este análisis, muestra aleatoria $X=(X_{1}, X_{2}, \ldots, X_{n})$ con $n=\text{tamaño de la muestra 1}$ y muestra aleatoria $Y=(Y_{1}, Y_{2}, \ldots, Y_{m})$ con $m=\text{tamaño de la muestra 2}$ corresponden a distribuciones normales, procedemos a determinar los coeficientes que las caracterizan (medias muestrales y desviaciones estándar, también se determinan los tamaños de las muestras). 

In [91]:
media_A = groupA.mean()
media_B = groupB.mean()
std_A = groupA.std()
std_B = groupB.std()
N_a = len(groupA)
N_b = len(groupB)
error_std_A = std_A/math.sqrt(N_a)
error_std_B = std_B/math.sqrt(N_b)
print('La media del grupo de varones es ', media_A)
print('La media del grupo de otros generos es ', media_B)
print('La desviación estandar del grupo de varones es ', std_A)
print('La desviación estandar del grupo de otros generos es ', std_B)
print('El número de muestras del grupo de varones es ', N_a)
print('El número de muestras del grupo de otros generos es ', N_b)
print('El error estandar del grupo de varones es ', error_std_A)
print('El error estandar del grupo de otros generos es ', error_std_B)


La media del grupo de varones es  101481.62434267912
La media del grupo de otros generos es  78219.25094276095
La desviación estandar del grupo de varones es  102215.30468444867
La desviación estandar del grupo de otros generos es  56515.32732350074
El número de muestras del grupo de varones es  4815
El número de muestras del grupo de otros generos es  891
El error estandar del grupo de varones es  1473.050995045699
El error estandar del grupo de otros generos es  1893.3347033956811


Como se observa, las dos muestras tienen medias, tamaños y desviaciones estándar diferentes. Esta información resultará relevante para la determinación del la estimación por intervalos.
A continuación se procede a calcular la estimación paramétrica puntual utiliando el test de Welch. Teniendo en cuenta la consigna, ésta corresponde a la diferencia de las medias de las muestras aleatorios objeto de análisis.

In [103]:
est_dif_medias = media_A - media_B
print('El estimador para la diferencia de medias es', est_dif_medias)

El estimador para la diferencia de medias es 23262.37339991817


A continuación se desarrolla la segunda parte de la consigna, un estimador por intervalos, específicamente un intervalo de confianza para la diferencia de medias muestrales. Como se mencionó, en el caso objeto de estudio las muestras $X\approx N(\mu_{1},\pmb{\sigma_{1}}^{2})$ e $Y\approx N(\mu_{2},\pmb{\pmb{\sigma}_{2}}^{2})$ se cumple que $\pmb{\sigma}_{1}\neq\pmb{\sigma}_{2}$, esto implica que para la determinación del estadístico se deberá emplear el Método de Welch. Esta último emplea una distribución t de sudent y el cálculo de los estadísticos que determinan los límites del intervalos de confianza implica conocer los grados de libertad de la distribución mencionada. En las siguientes lineas se calcula el estadístico y los grados de libertad de la distribución t de student para las muestras analizadas.

In [104]:
tmp = (media_A-media_B)/math.sqrt(error_std_A**2+error_std_B**2)
nu = ((std_A**2/N_a + std_B**2/N_b)**2)/(std_A**4/(N_a**2*(N_a-1))+std_B**4/(N_b**2*(N_b-1)))
print('El estadistico de prueba para el test de Welch es ', tmp)
print ('El numero de grados de libertad de t-student para el test de Welch es', nu)

El estadistico de prueba para el test de Welch es  9.697205976350673
El numero de grados de libertad de t-student para el test de Welch es 2148.0505828160863


Con un numero de grados de libertad tan grande, la distribución de t-student ya es una normal, por lo que se ustiliza este estadistico para una distribución normal. El valor de $t_{\frac{\alpha}{2},\nu-1}\sim z_{\frac{\alpha}{2}}$ para los valores $\alpha=0.05$, $\nu\sim2148$ esta dado por $z_{\frac{\alpha}{2}} = 1.96$, por lo que el intervalo de confianza estará dado por 
$(\bar{X}-\bar{Y})\pm1.96\sqrt(s_{\bar{X_1}^2}+s_{\bar{X_2}}^2)$

In [93]:
IC_min = est_dif_medias - 1.96*math.sqrt(error_std_A**2+error_std_B**2)
IC_max = est_dif_medias + 1.96*math.sqrt(error_std_A**2+error_std_B**2)
print('El intervalo de confianza 1-alpha es (',IC_min,',',IC_max,')')

El intervalo de confianza 1-alpha es ( 18560.580741802645 , 27964.166058033697 )


Teniendo en cuenta que se definió un nivel de significancia del 5% la $\underbrace{P}_{\text{Probabilidad}}\big((\theta_{1}-\theta_{2})\in(\hat{\theta}_{1}-\hat{\theta}_{2})\big)=1-\alpha=1-0.05=0.95$. Esto implicará que en el 95% de las realizaciones la diferencia de la media poblacional estará contenido en el intervalo encontrado.

A continuación, se utiliza scipy.stats para encontrar los $t_{\alpha,\nu-1}$ de la distribución t-student para calcular el intervalo de confianza y comparamos con los resultados obtenidos para una distribución normal

In [94]:
t_crit = np.abs(t.ppf(alpha/2,nu-1)) #este es el t_alpha,nu-1 de student 
print ('El valor del parametro t-student es ', t_crit)
IC_min_t = est_dif_medias - t_crit*math.sqrt(error_std_A**2+error_std_B**2)
IC_max_t = est_dif_medias + t_crit*math.sqrt(error_std_A**2+error_std_B**2)
print('El intervalo de confianza 1-alpha es (',IC_min_t,',',IC_max_t,')')

El valor del parametro t-student es  1.9610694947675884
El intervalo de confianza 1-alpha es ( 18558.01515881992 , 27966.731641016424 )


Observemos que, debido a que el numero de grados de libertad es tan grande, el valor del parametro de t-student es 1.96 aproximadamente y que el intervalo de confianza es practicamente indistinguible del calculado para una distribucion normal.

In [95]:
sp.stats.ttest_ind(groupA, groupB, equal_var=False, nan_policy='propagate', permutations=None, random_state=None, alternative='greater', trim=0)

Ttest_indResult(statistic=9.697205976350673, pvalue=4.302598661066604e-22)

## Ejercicio 2: Test de hipótesis



### 2.1 Formalización

Describir formalmente los distintos componentes de un test de hipótesis para comprobar si la distribución de los salarios es distinta entre los grupos A y B.

**Hipótesis Nula**

$H_0=...$

**Estadístico (Pivote)**
  * Identificar el estadístico
  * Escribir qué distribución tiene bajo $H_0$


La hipótesis nula sera que la diferencia de medias entre ambos grupos es cero, es decir $H_0:\mu_A-\mu_B=0$, y la hipótesis alternativa sera que $H_a:\mu_A-\mu_B>0$. Nuestro estadístico de prueba (pivote) estará dado por 
$$
T=\frac{\bar{X}-\bar{Y}}{\sqrt{s_{\bar{X}^2}+s_{\bar{Y}^2}}}
$$ 
que tiene distribución normal con media $\mu=0$ y desviación estandar $\sigma^2=1$ bajo $H_0$

El estadistico a utilizar dependera de la homogeneidad de varianzas, también conocido como supuesto de homocedasticidad, que considera 
que la varianza es constante (no varía) entre diferentes grupos(A y B). Cuando la condición de homocedasticidad no se cumple, se 
utiliza el término heterocedasticidad.
  

Existen diferentes test que permiten evaluar si las observaciones proceden de poblaciones con la misma varianza. Todos ellos consideran
como hipótesis nula que la varianza es igual entre los grupos y como hipótesis alternativa que no lo es. La diferencia entre ellos es el
estadístico de centralidad que utilizan:
​

- Los test que utilizan la media de la varianza son los que mayor poder estadístico consiguen cuando los datos siguen distribuciones normales.
- Los test que utilizan la mediana de la varianza consiguen mejores resultados cuando los datos siguen distribuciones asimétricas o se alejan 
de la normalidad.
​

Si no se puede alcanzar cierta seguridad de que las poblaciones que se comparan son de tipo normal, es recomendable recurrir a tests que comparen
la mediana de la varianza.
​

En primer lugar se procederá a aplicar el test de Levene, el test de Bartlett y el test de Fligner-Killeen  para determinar la homocedasticidad de
las varianzas de los grupos A y B.

#### Contraste de hipótesis

El test de Levene, el test de Bartlett y el test de Fligner-Killeen son tres de los test de hipótesis más empleados comparar la varianza entre grupos. 
En todos ellos, se considera como hipótesis nula que los datos proceden de distribuciones con la misma varianza (homocedasticidad). Por lo tanto, si 
el p-value es menor que un determinado valor (típicamente 0.05), entonces se considera que hay evidencias suficientes para rechazar la homocedasticidad
en favor de la heterocedasticidad.
​

El test de Levene y el test de Fligner-Killeen (este último no paramétrico) son más robustos que el test de Bartlett ante la falta de normalidad, por
lo que suele aconsejarse su uso. Además, tanto el test de Levene como el de Bartlett permiten elegir entre diferentes estadísticos de centralidad: 
mediana (por defecto), media, media truncada. Esto es importante a la hora de contrastar la homocedasticidad dependiendo de si los grupos se distribuyen
de forma normal o no.
​

Si se tiene seguridad de que las muestras a comparar proceden de poblaciones que siguen una distribución normal, es recomendable el test de Bartlett. 
Si no se tiene la seguridad de que las poblaciones de origen son normales, se recomiendan el test de Leven utilizando la mediana o el test no paramétrico
Fligner-Killeen que también se basa en la mediana.
  

In [96]:
# Levene test

levene_test = scipy.stats.levene(groupA, groupB, center='median')
print(levene_test)

LeveneResult(statistic=23.431491665834244, pvalue=1.3285372646647198e-06)


In [97]:
# Bartlett test

bartlett_test = scipy.stats.bartlett(groupA, groupB)
print(bartlett_test)

BartlettResult(statistic=400.59294156185115, pvalue=4.091277759372536e-89)


In [98]:
# Fligner test

fligner_test = scipy.stats.fligner(groupA, groupB, center='median')
print(fligner_test)

FlignerResult(statistic=38.31314850126406, pvalue=6.025528086005236e-10)


Todos  los test muestran evidencias para rechazar la hipótesis de que los dos grupos tienen la misma varianza, homocedasticidad. p-value < 0.05.

Por lo tanto el estadístico adecuado para la prueba de hipótesis de las diferencias de Medias es t de Welch.

### 2.2 P-valor

1. Calcule el p-valor y decida si rechazar o no la hipótesis nula.
2. Interprete el resultado.

Links útiles:
* [Test de hipótesis usando scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind.html)
* [Test de Welch](http://daniellakens.blogspot.com/2015/01/always-use-welchs-t-test-instead-of.html)

Como se indico anteriormente se utilizará la prueba t de Welch de forma predeterminada, ya que funciona mejor que la prueba t de Student siempre que los tamaños de muestra y las varianzas sean desiguales entre los grupos, y da el mismo resultado cuando los tamaños de muestra y las varianzas son iguales

**Prueba t de Welch**

La prueba t de Welch es una prueba univariante no paramétrica que prueba una diferencia significativa entre la media de dos grupos no relacionados. 

Es una alternativa a la prueba t de Student cuando se viola el supuesto de igualdad de varianzas.

**Supuestos de la prueba de t de welch**

Como toda prueba, esta prueba de estadística inferencial tiene suposiciones. Los supuestos que deben cumplir los datos para que los resultados de la prueba sean válidos son:

- La variable independiente  es categórica con al menos dos niveles (grupos A y B)

- La variable dependiente  es continua y se mide en una escala de intervalo o razón (salario)

- La distribución de los dos grupos debe seguir la distribución normal. (Se suponen distribuciones normales)

In [99]:
welch_test = scipy.stats.ttest_ind(groupA, groupB,nan_policy='omit', equal_var = False)
print(welch_test)

Ttest_indResult(statistic=9.697205976350673, pvalue=8.605197322133209e-22)


Ahora se utiliza el p-valor arriba calculado con scipy. Para un nivel de confianza $\alpha$, el p-valor se define como la probabilidad de obtener resultados al menos tan improbables o "extremos" como los observados en la muestra que tenemos, suponiendo que la hipótesis nula sea verdadera. 

Es decir, si tenemos una muestra de datos con un p-valor de 0,1, eso quiere decir que la probabilidad de haber obtenido los datos que tuvimos, siendo verdadera la hipótesis nula, es del 10%. Si anteriormente nos habiamos fijado una tolerancia o confianza de $\alpha=0.05$ o sea una tolerancia del 5%, eso quiere decir que nuestros datos son consistentes con la hipótesis nula y que esta no debe rechazarse en favor de la alternativa. En cambio, si por ejemplo tenemos un p-valor de 0.01, o sea un 1% de probabilidades de haber obtenido nuestros dados, debemos rechazar la hipótesis nula, ya que el p-valor esta por debajo de nuestra tolerancia. En resumen, si tenemos un nivel de confianza $\alpha$, la hipótesis nula no debe rechazarse si $\alpha$ es menor al p-valor; si por el contrario se da que $\alpha$ es mayor al p-valor, la hipótesis nula debe rechazarse en favor de la hipótesis alternativa. 

Observamos que nuestro p valor es $4.3\times10^{-22}$, es decir si las medias de los salarios de varones y otros géneros fueran iguales, los resultados que se obtuvieron en la encuesta se hubiesen obtenido con una probabilidad del $4.3\times10^{-22}$ (increiblemente baja!). Como nuestro $\alpha=0.05$, $\alpha$ es mayor (por veinte ordenes de magnitud) al p-valor, con lo cual se rechaza la hipótesis nula en favor de la alternativa, que es que los salarios del grupo A (varones) tienen una media mas alta que los del grupo B (otros generos).

### [Opcional] 2.3 Potencia del test

Nuestra muestra, ¿era lo suficientemente grande para detectar si existe o no una diferencia entre los grupos?

1. Utilice la función `tt_ind_solve_power` para calcular el tamaño necesario de la muestra para un poder estadístico de 0.8, 0.9 y 0.95, asumiendo una significancia estadística de 0.05.
2. ¿Cómo intepretan el poder estadístico de un test? Dado su conocimiento de dominio sobre los datos, ¿les parece que esta muestra es lo suficientemente grande para ser representativo de la tendencia general? ¿y para utilizarlo en un juicio penal contra una empresa XX por una causa de discriminación?

[Documentación](https://www.statsmodels.org/stable/generated/statsmodels.stats.power.tt_ind_solve_power.html)

NOTA: este análisis debería hacerse ANTES de recolectar los datos.

In [100]:
from statsmodels.stats.power import tt_ind_solve_power

In [101]:
effect_size = (groupA.mean() - groupB.mean()) / groupB.std()
# nobs1=None  - What we want to know
alpha = 0.05
ratio = len(groupB) / len(groupA)
power = [0.8,0.9,0.95]

In [102]:
for p in power:
    print('El tamaño necesario de las muestras para tener un poder estadístico de ', p, 'es', tt_ind_solve_power(effect_size=effect_size, alpha=alpha, power=p, ratio=ratio))

El tamaño necesario de las muestras para tener un poder estadístico de  0.8 es 298.3073379956589
El tamaño necesario de las muestras para tener un poder estadístico de  0.9 es 398.7961001403289
El tamaño necesario de las muestras para tener un poder estadístico de  0.95 es 492.8117730249174


El poder estadistico del test se define como la probabilidad de que el test de hipótesis nula este, efectivamente, rechazando correctamente la hipótesis nula. Es decir, si nuestra muestra tiene un poder estadistico de 0.8, quiere decir que se ha rechazado la hipótesis nula con un 80% de probabilidades de que este rechazo sea correcto. El test de poder estadistico nos da el tamaño que la muestra debe tener para que el test de hipótesis nula tenga el poder estadistico que le pedimos. Observemos que nos da que 

1. Para un poder estadístico del 80%, las muestras deben tener 298 elementos
2. Para un poder estadístico del 90%, las muestras deben tener 399 elementos
3. Para un poder estadístico del 95%, las muestras deben tener 493 elementos

Estos valores son el minimo que deben tener ambas muestras para que el test tenga la eficacia que le pedimos. Como el numero de muestras en el grupo de varones es 4815 y el numero de muestras en el grupo de otros generos es 891, vemos que ambas muestras superan ampliamente el minimo para todos los niveles de confianza requeridos. Podemos decir entonces que, por lo menos, el resultado obtenido con los test de hipótesis nula es correcto con una probabilidad del 95%. 

Podemos concluir que los resulatdos obtenidos son representativos de la tendencia general, ya que la muestra probo ser lo suficientemente grande para los niveles de confianza que exigimos.

## Ejercicio 3: Comunicación y visualización

**Consigna:** Seleccionen un resultado que les parezca relevante a partir de alguno de los ejercicios del entregable. Diseñe e implemente una comunicación en base a este mensaje, en un archivo PDF.

Elija las palabras y visualización más adecuada para que la comunicación sea entendible, efectiva y se dapte a UNA de las siguientes situaciones:

1. Una sección en un artículo de difusión a presentar como parte de una organización sin fines de lucro.
No más de 1 página A4 (o dos si los gráficos son muy grandes).
  1. Ejemplo: Alguna de las secciones [Los ecosistemas de emprendimiento de América Latina y el Caribe frente al COVID-19: Impactos, necesidades y recomendaciones](https://publications.iadb.org/es/los-ecosistemas-de-emprendimiento-de-america-latina-y-el-caribe-frente-al-covid-19-impactos-necesidades-y-recomendaciones), por ejemplo la sección *2.2. Reacciones de los emprendedores*.
  2. Ejemplo: Alguna de las secciones de [The state of gender pay gap in 2021](https://www.payscale.com/data/gender-pay-gap?tk=carousel-ps-rc-job)
  3. Puntos clave: 
    1. Simpleza de los gráficos.
    2. Comunicación en lenguaje simple a personas que no son necesariamente expertos de dominio.
    2. Selección de UNA oración sobre la que se hace énfasis.
    3. No es necesario que mencionen objetivos ni descripciones del conjunto de datos, se supone que eso ya estaría explicado en otras secciones del informe.

2. Una publicación científica o reporte técnico interno. No más de una página A4:
  2. Ejemplo: La sección de resultados de [IZA DP No. 12914: The Impact of a Minimum Wage Change on the Distribution of Wages and Household Income](https://www.iza.org/publications/dp/12914/the-impact-of-a-minimum-wage-change-on-the-distribution-of-wages-and-household-income).
  2. Ejemplo: Alguna de las secciones de [Temporary reduction in daily global CO2 emissions during the COVID-19 forced confinement](https://www.nature.com/articles/s41558-020-0797-x)
  3. Puntos clave:
    3. Nivel de detalle técnico requerido. Es necesario justificar la validez del análisis.
    4. La idea presentada puede ser más compleja. Pueden asumir que la audiencia tiene conocimiento técnico y va a analizar las visualizaciones en detalle. 
    5. Pueden presentar más en detalle las limitaciones del análisis (significancia estadística, etc.)
    2. No es necesario que mencionen objetivos ni descripciones del conjunto de datos, se supone que eso ya estaría explicado en otras secciones del informe.

3. Un tweet (o post de LinkedIn) para la cuenta de su empresa consultora que hace análisis de datos. El objetivo es promocionar un análisis de datos abiertos que van a incluir en su portfolio:
  1. Ejemplo: [Comparación vacunas covid](https://twitter.com/infobeautiful/status/1381577746527236098?s=20)
  2. Ejemplo: [Tweet del BID](https://twitter.com/el_BID/status/1388508583944507396?s=20). Lo valioso de este tweet es que usaron un único número para transmitir un mensaje. Puede ser algo así, o con un gráfico muy simple.
  3. Ejemplo: [Cambio climático](https://twitter.com/UNFCCC/status/1387732156190011394?s=20) Es un muy buen ejemplo, excepto que el gráfico no se lee nada y hay que entrar a la publicación original.
  3. Ejemplo: [¿Cuánto están los programadores en las empresas?](https://www.linkedin.com/posts/denis-rothman-0b034043_tech-career-work-activity-6793861923269054464-gS6y) (No verificamos la veracidad o seriedad de la fuente).
  4. Puntos clave:
    1. Su audiencia no va a mirar la visualización por más de unos segundos, y no tiene conocimiento técnico.
  3. Tienen que incluir además una *breve* descripción de cómo obtuvieron los datos que están presentando, que no entraría en el tweet.
